sound/hda/intel-dsp-config.c

```diff
@@ -368,7 +396,12 @@ int snd_intel_dsp_driver_probe(struct pci_dev *pci)
 		return SND_INTEL_DSP_DRIVER_ANY;
 
 	if (cfg->flags & FLAG_SOF) {
-		if (cfg->flags & FLAG_SOF_ONLY_IF_DMIC) {
+		if (cfg->flags & FLAG_SOF_ONLY_IF_SOUNDWIRE) {
+			if (snd_intel_dsp_check_soundwire(pci) > 0) {
+				dev_info(&pci->dev, "SoundWire enabled on CannonLake+ platform, using SOF driver\n");
+				return SND_INTEL_DSP_DRIVER_SOF;
+			}
```

Comment:
@plbossart just wondering what happens in the else case here?

Reply:
I need more excedrin

Reply:
fixed now